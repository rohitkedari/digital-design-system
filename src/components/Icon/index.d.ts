import React from 'react';
import FontAwesomeIconProps from '@fortawesome/react-fontawesome';

export interface IconProps extends FontAwesomeIconProps {}
declare const Icon: React.SFC<IconProps>;

export { Icon };
