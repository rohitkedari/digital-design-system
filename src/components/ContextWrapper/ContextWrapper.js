/**
 * context wrapper component
 */
import React from 'react';
import PropTypes from 'prop-types';
import { ThemeContext, ResponsiveContextProvider } from 'contexts';
import { deepMerge, GlobalStyle, loadFonts } from 'utils';
import { baseTheme } from 'themes/base';

const ContextWrapper = ({ children, theme }) => {
    const nextTheme = deepMerge(baseTheme, theme);
    loadFonts(nextTheme.global.font.url);
    return (
        <ThemeContext.Provider value={nextTheme}>
            <ResponsiveContextProvider
                breakpoints={nextTheme.global.breakpoints}
            >
                <GlobalStyle />
                {children}
            </ResponsiveContextProvider>
        </ThemeContext.Provider>
    );
};
ContextWrapper.defaultProps = {
    theme: null,
};
ContextWrapper.propTypes = {
    children: PropTypes.node.isRequired,
    theme: PropTypes.shape({}),
};
export { ContextWrapper };
