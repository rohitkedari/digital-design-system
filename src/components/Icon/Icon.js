import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { getColor } from 'utils';

const StyledIcon = styled(FontAwesomeIcon)`
	${(props) => props.color && `color: ${getColor(props.color, props.theme)};`}
`;
const Icon = ({ ...rest }) => {
	return <StyledIcon {...rest} />;
};

Icon.defaultProps = {
	border: false,
	mask: null,
	fixedWidth: false,
	inverse: false,
	flip: null,
	icon: null,
	listItem: false,
	pull: null,
	pulse: false,
	rotation: null,
	size: null,
	spin: false,
	swapOpacity: false,
	symbol: false,
	title: '',
	transform: null,
};

Icon.propTypes = {
	border: PropTypes.bool,
	fixedWidth: PropTypes.bool,
	flip: PropTypes.oneOf(['horizontal', 'vertical', 'both']),
	icon: PropTypes.oneOfType([
		PropTypes.object,
		PropTypes.array,
		PropTypes.string,
	]),
	inverse: PropTypes.bool,
	listItem: PropTypes.bool,
	mask: PropTypes.oneOfType([
		PropTypes.object,
		PropTypes.array,
		PropTypes.string,
	]),
	pull: PropTypes.oneOf(['right', 'left']),
	pulse: PropTypes.bool,
	rotation: PropTypes.oneOf([0, 90, 180, 270]),
	size: PropTypes.oneOf([
		'lg',
		'xs',
		'sm',
		'1x',
		'2x',
		'3x',
		'4x',
		'5x',
		'6x',
		'7x',
		'8x',
		'9x',
		'10x',
	]),
	spin: PropTypes.bool,
	swapOpacity: PropTypes.bool,
	symbol: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
	title: PropTypes.string,
	transform: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

export { Icon };
