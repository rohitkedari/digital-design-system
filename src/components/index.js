/* GENERATOR: Assemble all components for export */
export * from './Modal';
export * from './Select';
export * from './ContextWrapper';
export * from './Heading';
export * from './Paragraph';
export * from './Icon';
