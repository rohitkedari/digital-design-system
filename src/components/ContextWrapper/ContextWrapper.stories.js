import React, { useContext } from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import { ThemeContext, ResponsiveContext } from 'contexts';
import { withA11y } from '@storybook/addon-a11y';
import { ContextWrapper } from './ContextWrapper';

const StyledContent = styled('div')`
	color: ${(props) => props.color || props.theme.global.colors.brand};
`;
storiesOf('components|ContextWrapper', module)
	.addParameters({ component: ContextWrapper })
	.addDecorator(withA11y)
	.add('Default Context', () => {
		const Content = () => {
			const { viewport } = useContext(ResponsiveContext);
			const theme = useContext(ThemeContext);
			return (
				<div>
					{`This is a ${viewport} screen`}
					<StyledContent>Text with brand color</StyledContent>
					<StyledContent color={theme.global.colors.status_ok}>
						Text with success color
					</StyledContent>
					<StyledContent color={theme.global.colors.status_error}>
						Text with error color
					</StyledContent>
					<StyledContent color={theme.global.colors.status_warning}>
						Text with warning color
					</StyledContent>
				</div>
			);
		};
		return (
			<ContextWrapper>
				<Content />
			</ContextWrapper>
		);
	});
