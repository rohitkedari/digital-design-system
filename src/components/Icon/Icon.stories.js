import React from 'react';
import { storiesOf } from '@storybook/react';
import { withTests } from '@storybook/addon-jest';
import { faPhone, faSpinner } from '@fortawesome/free-solid-svg-icons';
import { Icon } from './Icon';
import results from './__tests__/Icon-test-results.json';

storiesOf('components|Icon', module)
	.addDecorator(
		withTests({
			results,
		}),
	)
	.addParameters({ component: Icon, jest: ['Icon.test.js'] })
	.add('Default', () => <Icon icon={faPhone} title="coffee" />)
	.add('With Size', () => (
		<>
			<Icon icon={faPhone} title="coffee" size="xs" />
			<Icon icon={faPhone} title="coffee" size="sm" />
			<Icon icon={faPhone} title="coffee" size="lg" />
		</>
	))
	.add('With Spin', () => <Icon icon={faSpinner} title="loading" spin />)
	.add('With Color', () => (
		<>
			<Icon icon={faPhone} title="loading" color="red" />
			<Icon icon={faPhone} title="loading" color="brand" />
			<Icon icon={faPhone} title="loading" color="#66BF3C" />
		</>
	))
	.add('With Border', () => (
		<Icon icon={faPhone} title="loading" border size="4x" />
	));
