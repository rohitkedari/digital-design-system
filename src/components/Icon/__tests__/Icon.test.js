import React from 'react';
import renderer from 'react-test-renderer';
import { ContextWrapper } from 'components/ContextWrapper';
import { faPhone, faSpinner } from '@fortawesome/free-solid-svg-icons';
import { Icon } from '../Icon';

describe('Icon', () => {
	beforeAll(() =>
		jest.spyOn(global.Math, 'random').mockImplementation(() => 0),
	);
	afterAll(() => jest.restoreAllMocks());
	test('default renders correctly', () => {
		const tree = renderer
			.create(
				<ContextWrapper>
					<Icon icon={faPhone} title="phone icon" />
				</ContextWrapper>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('with size renders correctly', () => {
		const tree = renderer
			.create(
				<ContextWrapper>
					<Icon icon={faPhone} title="phone icon" size="xs" />
					<Icon icon={faPhone} title="phone icon" size="sm" />
					<Icon icon={faPhone} title="phone icon" size="lg" />
				</ContextWrapper>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('with spin renders correctly', () => {
		const tree = renderer
			.create(
				<ContextWrapper>
					<Icon icon={faSpinner} title="loading" spin />
				</ContextWrapper>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('with color renders correctly', () => {
		const tree = renderer
			.create(
				<ContextWrapper>
					<Icon icon={faPhone} title="red phone icon" color="red" />
					<Icon
						icon={faPhone}
						title="brand phone icon"
						color="brand"
					/>
					<Icon
						icon={faPhone}
						title="green phone icon"
						color="#66BF3C"
					/>
				</ContextWrapper>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('with border renders correctly', () => {
		const tree = renderer
			.create(
				<ContextWrapper>
					<Icon
						icon={faPhone}
						title="phone with border"
						border
						size="4x"
					/>
				</ContextWrapper>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
