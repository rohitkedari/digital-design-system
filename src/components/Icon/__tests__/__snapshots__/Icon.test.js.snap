// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon default renders correctly 1`] = `
<svg
  aria-labelledby="svg-inline--fa-title-000000000000"
  className="svg-inline--fa fa-phone fa-w-16 sc-AxjAm jLWHxH"
  data-icon="phone"
  data-prefix="fas"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <title
    id="svg-inline--fa-title-000000000000"
    style={Object {}}
  >
    phone icon
  </title>
  <path
    d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`Icon with border renders correctly 1`] = `
<svg
  aria-labelledby="svg-inline--fa-title-000000000000"
  className="svg-inline--fa fa-phone fa-w-16 fa-border fa-4x sc-AxjAm jLWHxH"
  data-icon="phone"
  data-prefix="fas"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <title
    id="svg-inline--fa-title-000000000000"
    style={Object {}}
  >
    phone with border
  </title>
  <path
    d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`Icon with color renders correctly 1`] = `
Array [
  <svg
    aria-labelledby="svg-inline--fa-title-000000000000"
    className="svg-inline--fa fa-phone fa-w-16 sc-AxjAm gUTuSI"
    color="red"
    data-icon="phone"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="svg-inline--fa-title-000000000000"
      style={Object {}}
    >
      red phone icon
    </title>
    <path
      d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>,
  <svg
    aria-labelledby="svg-inline--fa-title-000000000000"
    className="svg-inline--fa fa-phone fa-w-16 sc-AxjAm eBJTpR"
    color="brand"
    data-icon="phone"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="svg-inline--fa-title-000000000000"
      style={Object {}}
    >
      brand phone icon
    </title>
    <path
      d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>,
  <svg
    aria-labelledby="svg-inline--fa-title-000000000000"
    className="svg-inline--fa fa-phone fa-w-16 sc-AxjAm laZens"
    color="#66BF3C"
    data-icon="phone"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="svg-inline--fa-title-000000000000"
      style={Object {}}
    >
      green phone icon
    </title>
    <path
      d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>,
]
`;

exports[`Icon with size renders correctly 1`] = `
Array [
  <svg
    aria-labelledby="svg-inline--fa-title-000000000000"
    className="svg-inline--fa fa-phone fa-w-16 fa-xs sc-AxjAm jLWHxH"
    data-icon="phone"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="svg-inline--fa-title-000000000000"
      style={Object {}}
    >
      phone icon
    </title>
    <path
      d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>,
  <svg
    aria-labelledby="svg-inline--fa-title-000000000000"
    className="svg-inline--fa fa-phone fa-w-16 fa-sm sc-AxjAm jLWHxH"
    data-icon="phone"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="svg-inline--fa-title-000000000000"
      style={Object {}}
    >
      phone icon
    </title>
    <path
      d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>,
  <svg
    aria-labelledby="svg-inline--fa-title-000000000000"
    className="svg-inline--fa fa-phone fa-w-16 fa-lg sc-AxjAm jLWHxH"
    data-icon="phone"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="svg-inline--fa-title-000000000000"
      style={Object {}}
    >
      phone icon
    </title>
    <path
      d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>,
]
`;

exports[`Icon with spin renders correctly 1`] = `
<svg
  aria-labelledby="svg-inline--fa-title-000000000000"
  className="svg-inline--fa fa-spinner fa-w-16 fa-spin sc-AxjAm jLWHxH"
  data-icon="spinner"
  data-prefix="fas"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <title
    id="svg-inline--fa-title-000000000000"
    style={Object {}}
  >
    loading
  </title>
  <path
    d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;
